data/interfaces/default/newsletter_config.html

```diff
@@ -471,8 +486,201 @@ <h4 class="modal-title" id="newsletter-config-modal-header">${newsletter['agent_
 </div>
 
 <script src="${http_root}js/jquery-cron-min.js"></script>
+<script src="${http_root}js/ace/ace.js"></script>
+<script src="${http_root}js/theme-tautulli.js"></script>
+<script src="${http_root}js/beautify-html.min.js"></script>
 <script>
 
+    // Initialize Ace Editor for the message field
+    var messageEditor;
+    
+    function initMessageEditor() {
+        if (messageEditor) {
+            messageEditor.destroy();
+        }
+        
+        try {
+            // Check if Ace editor is available
+            if (typeof ace === 'undefined') {
+                throw new Error('Ace editor not loaded');
+            }
+            
+            messageEditor = ace.edit('message-editor');
+            messageEditor.setTheme('ace/theme/tautulli');
+            messageEditor.session.setMode('ace/mode/html');
+            messageEditor.setOptions({
+                showPrintMargin: false,
+                showGutter: true,
+                fontSize: '14px',
+                tabSize: 2,
+                wrap: true,
+                highlightActiveLine: true
+            });
+            
+            // Set initial value from the hidden textarea
+            var initialValue = $('#message').val() || '';
+            messageEditor.setValue(initialValue);
+            messageEditor.clearSelection();
+            
+            // Sync Ace editor content with hidden textarea on change
+            messageEditor.getSession().on('change', function() {
+                $('#message').val(messageEditor.getValue());
+            });
```

Comment:
Filter out HTML annotations with the HTML helper.

```suggestion
            var session = messageEditor.getSession();
            
            // Sync Ace editor content with hidden textarea on change
            session.on('change', function() {
                $('#message').val(messageEditor.getValue());
            });
            
            // Filter out HTML annotations
            session.on('changeAnnotation', function () {
                session.$annotations = session.$annotations.filter(function(annotation){
                    return !(/doctype first\. Expected/.test(annotation.text) || /Unexpected End of file\. Expected/.test(annotation.text))
                });
                messageEditor.$onChangeAnnotation();
            });
```